docs/k8s-cluster/nvidia-network-operator.md

```diff
@@ -0,0 +1,233 @@
+
+     ```sh
+     # NOTE: If SSH requires a password, add: `-k`
+     # NOTE: If sudo on remote machine requires a password, add: `-K`
+     # NOTE: If SSH user is different than current user, add: `-u <user>`
+     ansible-playbook -b playbooks/generic/users.yml
+     ```
+     Verify the configuration
+
+     ```sh
+     ansible all -m raw -a "hostname"
+     ```
+
+  - Deploying and verifying Kubernetes cluster
+    Install Kubernetes using Ansible and Kubespray
+
+    ```sh
+    # NOTE: If SSH requires a password, add: `-k`
+    # NOTE: If sudo on remote machine requires a password, add: `-K`
+    # NOTE: If SSH user is different than current user, add: `-u ubuntu`
+    ansible-playbook -l k8s-cluster playbooks/k8s-cluster.yml
+    ```
+    Please refer to [DeepOps Kubernetes Deployment Guidehere](https://github.com/NVIDIA/deepops/blob/master/docs/kubernetes-cluster.md) for more information.
+    
+    Verify that Kubernetes clustering is working with "kubectl get nodes" command:
+    ```sh
+    nvidia@mgmt01:~$ kubectl get nodes
+    NAME     STATUS   ROLES    AGE    VERSION
+    mgmt01   Ready    master   1d8h   v1.21.6
+    gpu01    Ready    <none>   1d8h   v1.21.6
+    gpu02    Ready    <none>   1d8h   v1.21.6
+    nvidia@mgmt01:~$
+    ```
+5. Deploy Nvidia Network Operator
+   Before runnng the playbook, please update "roles/nvidia-network-operator/vars/main.yml" file according to your hardware and network configuration, this is what we used in our value.yaml file:
+   ```sh
+   num_vf: 8
+   vendor_id: "15b3"
+   link_type: "ib"
+   mtu: 4096
+
+   intf_resources:
+     - if_name: "ibs1"
+       pf_name: "ibs1"
+       res_name: "resibs1"
+       ip_addr: "192.168.101.0/24"
+     - if_name: "ibp12s0"
+       pf_name: "ibp12s0"
+       res_name: "resibp12s0"
+       ip_addr: "192.168.102.0/24"
+       ...
+    ## "15b3" is Mellanox vendor code for ConnectX cards.
+    ```
+   Run the playbook:
+    ```sh
+   ansible-playbook playbooks/k8s-cluster/nvidia-network-operator.yaml
+   ```
+   
+## Running the Workload
+
+The cluster is ready to run multi-node workload in the cluster, One last thing is to add related interface configuration to the job file before launching your job. 
+
+### Using SRIOV interfaces
+
+Below is what is the section of the job file looks like after adding relevant SRIOV interface configuration. We built a docker private registry at 192.168.1.11 to host and manage our testing images, please refer to this docker [document](https://docs.docker.com/registry/deploying/) for more details.
+
+```sh
+Worker:
+  replicas: 2
+  template:
+    metadata:
+      annotations:
+        k8s.v1.cni.cncf.io/networks: ibs1,ibp12s0
+    spec:
+      containers:
+      - image: 192.168.1.11:5000/nccl-test
+        name: nccl-benchmark
+        securityContext:
+          capabilities:
+            add: [ "IPC_LOCK" ]
+        resources:
+          limits:
+            nvidia.com/resibs1: "1"

```

Comment:
Is this added by the network operator? Can we add a section describing this or add a link to docs?